### PR TITLE
[prompty] Prompty supports chat history

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
Prompty supports chat history

pf flow test --flow path/to/prompty.prompty --interactive
image

```
---
name: Basic Prompt With Chat History
description: A basic prompt that uses the GPT-3 chat API to answer questions
model:
    api: chat
    configuration:
      type: azure_open_ai
      azure_deployment: gpt-35-turbo
      connection: azure_open_ai_connection

    parameters:
      max_tokens: 128
      temperature: 0.2
inputs:
  firstName:
    type: string
    default: John
  lastName:
    type: string
    default: Doh
  question:
    type: string
    is_chat_input: true
  chat_history:
    is_chat_history: true
    type: list
    default: [
        {
          "role": "user",
          "content": "what is the result of 1+1?"
        },
        {
          "role": "Assistant",
          "content": "The result of 1+1 is 2. It's a simple addition!"
        },
        {
          "role": "user",
          "content": "what is the result of 2+2?"
        },
        {
          "role": "Assistant",
          "content": "The result of 2+2 is 4. It's a simple addition!"
        }
    ]
---
system:
You are an AI assistant who helps people find information.
As the assistant, you answer questions briefly, succinctly,
and in a personable manner using markdown and even add some personal flair with appropriate emojis.

# Safety
- You **should always** reference factual statements to search results based on [relevant documents]
- Search results based on [relevant documents] may be incomplete or irrelevant. You do not make assumptions
# Customer
You are helping {{firstName}} {{lastName}} to find answers to their questions.
Use their name to address them in your responses.

Here is a chat history you had with the user:
{% for item in chat_history %}
   {{item.role}}: {{item.content}}
{% endfor %}

user:
{{question}}

```

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py [hunk 1]
```diff
@@ -1045,6 +1045,27 @@ def mock_input(*args, **kwargs):
         outerr = capsys.readouterr()
         assert "chat flow does not support multiple chat outputs" in outerr.out
 
+        # Test prompty with stream output
```
Comment (crazygao) [inline src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py:1048]: speed up tests by creating more functions...

File: src/promptflow-core/promptflow/_utils/flow_utils.py [hunk 2]
```diff
@@ -288,9 +288,12 @@ def is_executable_chat_flow(flow: ExecutableFlow):
     if len(chat_inputs) != 1:
         _is_chat_flow = False
         error_msg = "chat flow does not support multiple chat inputs"
-    elif len(chat_outputs) != 1:
+    elif len(chat_outputs) > 1:
         _is_chat_flow = False
         error_msg = "chat flow does not support multiple chat outputs"
+    elif not chat_outputs and len(flow.outputs.values()) > 0:
+        _is_chat_flow = False
+        error_msg = "chat output is not configured"
```
Comment (crazygao) [inline src/promptflow-core/promptflow/_utils/flow_utils.py:296]: create negative tests for these error msgs?

File: src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow-devkit/tests/sdk_cli_test/e2etests/test_cli.py:1090]: Split these test case

File: src/promptflow-core/promptflow/_utils/flow_utils.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow-core/promptflow/_utils/flow_utils.py:296]: add megtive test cases

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py [hunk 3]
```diff
@@ -111,6 +112,8 @@ def flow(self):
     @property
     def dataplane_flow(self):
         # TODO: test submitter shouldn't interact with dataplane flow directly
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py:114]: looks like we will always do a deserialize? check not _data_plane_flow here also

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py [hunk 4]
```diff
@@ -546,24 +551,50 @@ def change_logger_level(level):
             with change_logger_level(level=logging.WARNING):
                 chat_inputs, _ = self.resolve_data(inputs=inputs)
 
+            init_kwargs = None
+            if isinstance(self.flow, Prompty):
+                # Override prompt output format configuration and only return first choice in interactive mode.
+                init_kwargs = {"model": {"parameters": {"stream": False}, "response": "first"}}
```
Comment (wangchao1230) [inline src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py:557]: why set this to stream False? I remember we enabled streaming in interactive mode

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py (hunk 3 above)
Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py:114]: fix it

File: src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py (hunk 4 above)
Comment (lalala123123) [reply src/promptflow-devkit/promptflow/_sdk/_orchestrator/test_submitter.py]: revert it. no need to set stream to false

**Review by Adrian999999999 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Pull request merged**